File: branches/Andean/Fantasy.Base/ServiceModel/ServiceProviderExtension.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.ServiceModel;

namespace Fantasy
{
    public static class ServiceProviderExtension
    {

        public static T GetService<T>(this IServiceProvider services)
        {
            return (T)services.GetService(typeof(T)); 
        }

        public static object GetRequiredService(this IServiceProvider services, Type serviceType)
        {
            object rs = services.GetService(serviceType);
            if (rs == null)
            {
                throw new MissingRequiredServiceException(serviceType); 
            }
            return rs;
        }

        public static T GetRequiredService<T>(this IServiceProvider services)
        {
            return (T)services.GetRequiredService(typeof(T)); 
        }


    }
}

File: branches/Stheno/AutoConverted/Fantasy.Base/ServiceModel/ServiceProviderExtension.java
package Fantasy;

import Fantasy.ServiceModel.*;

public final class ServiceProviderExtension
{

//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static T GetService<T>(this IServiceProvider services)
	public static <T> T GetService(IServiceProvider services)
	{
		return (T)services.GetService(T.class);
	}

//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static object GetRequiredService(this IServiceProvider services, Type serviceType)
	public static Object GetRequiredService(IServiceProvider services, java.lang.Class serviceType)
	{
		Object rs = services.GetService(serviceType);
		if (rs == null)
		{
			throw new MissingRequiredServiceException(serviceType);
		}
		return rs;
	}

//C# TO JAVA CONVERTER TODO TASK: Extension methods are not available in Java:
//ORIGINAL LINE: public static T GetRequiredService<T>(this IServiceProvider services)
	public static <T> T GetRequiredService(IServiceProvider services)
	{
		return (T)services.GetRequiredService(T.class);
	}


}